### PR TITLE
remove ipi-deprovision tool and its related image 

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/openshift-team-developer-productivity-test-platform 

This tool is deprecated. See [here](https://github.com/openshift/release/tree/master/core-services/ipi-deprovision). 
This PR also removes the script from the ipi-deprovision image.

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

File: images/ipi-deprovision/Dockerfile [hunk 1]
```diff
@@ -5,18 +5,12 @@ ARG JQ_VERSION=1.6
```
Comment (alvaroaleman) [inline images/ipi-deprovision/Dockerfile:5]: Why do we need to keep this image here, shouldn't we simply make this a BuildConfig instead?

Comment (alvaroaleman) [reply images/ipi-deprovision/Dockerfile:5]: All imagestreamtags in the CI get automatically distributed to all build clusters already: https://github.com/openshift/release/blob/8568b711c7374cd765c75d3086c92140e21a49d8/clusters/app.ci/assets/dptp-controller-manager.yaml#L126

So if you just put it there, nothing more to do.

Comment (droslean) [reply images/ipi-deprovision/Dockerfile:5]: Also, there are 2 repo files that this image is using. I am not sure if we can inject them in the buildConfig.

Comment (droslean) [reply images/ipi-deprovision/Dockerfile:5]: Let's take this approach for now. I will open a card to completely remove that image and use a bc.

Comment (alvaroaleman) [reply images/ipi-deprovision/Dockerfile:5]: > Also, there are 2 repo files that this image is using. I am not sure if we can inject them in the buildConfig.

Yes, you can use `echo` in a `docker run` directive, I would assume a heredoc works as well. This image has literally nothing to do with this repo so I would very much prefer if we didn't build it on every PR here.

Comment (droslean) [reply images/ipi-deprovision/Dockerfile:5]: https://github.com/openshift/release/pull/17065

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**